develop/dev-guide-timeouts-in-tidb.md

```diff
@@ -11,9 +11,38 @@ This document describes various timeouts in TiDB to help you troubleshoot errors
 
 TiDB's transaction implementation uses the MVCC (Multiple Version Concurrency Control) mechanism. When the newly written data overwrites the old data, the old data will not be replaced, but kept together with the newly written data. The versions are distinguished by the timestamp. TiDB uses the mechanism of periodic Garbage Collection (GC) to clean up the old data that is no longer needed.
 
-By default, each MVCC version (consistency snapshots) is kept for 10 minutes. Transactions that take longer than 10 minutes to read will receive an error `GC life time is shorter than transaction duration`.
+- For versions of TiDB prior to `4.0`:
 
-If you need longer read time, for example, when you are using **Dumpling** for full backups (**Dumpling** backs up consistent snapshots), you can adjust the value of `tikv_gc_life_time` in the `mysql.tidb` table in TiDB to increase the MVCC version retention time. Note that `tikv_gc_life_time` takes effect globally and immediately. Increasing the value will increase the life time of all existing snapshots, and decreasing it will immediately shorten the life time of all snapshots. Too many MVCC versions will impact TiKV's processing efficiency. So you need to change `tikv_gc_life_time` back to the previous setting in time after doing a full backup with **Dumpling**.
+    By default, each MVCC version (consistency snapshots) is kept for 10 minutes. Transactions that take longer than 10 minutes to read will receive an error `GC life time is shorter than transaction duration`.
+
+- For TiDB version 4.0 and later:
+
+    For transactions that are active and do not exceed a duration of 24 hours, garbage collection (GC) will be blocked during the transaction period, preventing the occurrence of the error `GC life time is shorter than transaction duration`.
+
+If you need longer read time temporarily, you can adjust the value of:
+
+- For versions of TiDB prior to `5.0`:
+
+    The `tikv_gc_life_time` in the `mysql.tidb` table in TiDB
+
+- For TiDB version 5.0 and later:
+
+    The system variable [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-new-in-v50)
+
+To increase the MVCC version retention time. Note that this variable takes effect globally and immediately. Increasing the value will increase the life time of all existing snapshots, and decreasing it will immediately shorten the life time of all snapshots. Too many MVCC versions will impact the performance of TiDB cluster. So you need to change this variable back to the previous setting in time.
```

Comment:
```suggestion
Note that the system variable configuration takes effect globally and immediately. Increasing its value will increase the life time of all existing snapshots, and decreasing it will immediately shorten the life time of all snapshots. Too many MVCC versions will impact the performance of the TiDB cluster. So you need to change this variable back to the previous setting in time.
```